specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/1.0.0/iotcentral.json

```diff
@@ -0,0 +1,2262 @@
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceTemplate"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "DeviceTemplates_Remove",
+        "summary": "Delete a device template",
+        "description": "Delete an existing device template by device ID.",
+        "x-ms-examples": {
+          "Delete a device template by Id": {
+            "$ref": "./examples/devicetemplates_delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceTemplateId"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Success"
+          }
+        }
+      }
+    },
+    "/devices": {
+      "get": {
+        "operationId": "Devices_List",
+        "summary": "Get the list of devices in an application",
+        "x-ms-examples": {
+          "List devices": {
+            "$ref": "./examples/devices_list.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DeviceCollection"
+            }
+          }
+        }
+      }
+    },
+    "/devices/{device_id}": {
+      "get": {
+        "operationId": "Devices_Get",
+        "summary": "Get a device by ID",
+        "description": "Get details about an existing device by device ID.",
+        "x-ms-examples": {
+          "Get device by Id": {
+            "$ref": "./examples/devices_get.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "Devices_Create",
+        "summary": "Create or update a device",
+        "description": "Create a new device.",
+        "x-ms-examples": {
+          "Create device": {
+            "$ref": "./examples/devices_create.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/DeviceId"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Device body.",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Device"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "Devices_Update",
```

Comment:
Is this a true patch, meaning I can pass a partial payload?

If so, I wonder about using the word "update" here and whether that is standard. I could be wrong, but I've taken update to mean full payload over PUT ideally using an ETag.

Reply:
@johanste any opinions on this one?

Reply:
A patch should specify which patch format/specification it follows by providing the correct mime type in the `consumes`. If what you have completely replaces the object, it should be a PUT operation. Does the PUT fail if the resource doesn't already exist?

Reply:
This operation id was updated from preview when we introduce the PATCH, it support partial payload, we also update the operation id PUT to Create, but even for PUT, if resource existed we will replace the existed one

Reply:
For PUT, create or replace is expected. Which content type do you accept for PATCH?

Reply:
we use content type: application/json

Reply:
Not application/json-patch+json?

Here's an example in [ADT](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/digitaltwins/data-plane/Microsoft.DigitalTwins/stable/2020-10-31/digitaltwins.json#L184).

Reply:
Thanks @drwill-ms, this really helps, will make the changes.

Reply:
Added support content type, as backward compatibility, I enabled both.